arch/Kconfig

```diff
@@ -445,6 +445,7 @@ menuconfig ARCH_ADDRENV
 	bool "Address environments"
 	default n
 	depends on ARCH_HAVE_ADDRENV
+	depends on ARCH_INTERRUPTSTACK != 0
```

Comment:
per task kernel stack also work?

Reply:
but "depends on ARCH_KERNEL_STACK || ARCH_INTERRUPTSTACK != 0" is more accurate.

Reply:
True, however the infrastructure for using ARCH_KERNEL_STACK for this is not in place for any of the targets. So it can be a bit misleading IMO.

Reply:
Once it's implemented, the problem is resolved naturally, why not relax this condition at the beginning? I am afraid the developer who implement the kernel stack forget to change the condition, and other people who doesn't familiar the internal detail misunderstand that they can't remove the interrupt stack even the per task kernel stack is enabled.

Reply:
Sure, I'll add it here

Reply:
look like we have the circle dependence:(

Reply:
Yes but of course. The per process kernel stack requires an address environment. The intent of this commit was to add a guard against a misconfiguration as group_addrenv() cannot be executed on the user stack.

But I guess we are left with these options:
- Require the IRQ stack, which is not really true
- Abandon this change, and hope future users of ARCH_ADDRENV realize that a neutral stack is needed
- Something else ?

Reply:
let's add static_assert in exception_common.S?

Reply:
Yeah makes sense, I'll do that instead